blocks/dfio.py

```diff
@@ -140,9 +149,11 @@ def _write_avro(df, handle, **write_args):
 def _get_extension(path):
     name, ext = os.path.splitext(path)
     # Support compression extensions, eg part.csv.gz
+    comp = None
     if ext in _compressions and '.' in name:
-        return _get_extension(name)
-    return ext
+        comp = ext
+        name, ext = os.path.splitext(name)
+    return ext, comp
```

Comment:
comp needs to be _compression.get(comp), and we need to change
_compressions = {
    '.gz': 'gzip',
    '.bz2': 'bz2',
    '.zip': 'zip',
    '.xz': 'xz'
}
because the compression kwargs are different than the extension. Forgot to copy that part over

Reply:
Added - it turns out we also need to manually gzip the csv at write time. Pandas supports unzipping file handles at read but not zipping them at right (only supports zipping when provided a file path).